src/libstd/thread/mod.rs

```diff
@@ -806,9 +806,13 @@ const NOTIFIED: usize = 2;
 /// In other words, each [`Thread`] acts a bit like a spinlock that can be
 /// locked and unlocked using `park` and `unpark`.
 ///
+/// Notice that it would be a valid (but inefficient) implementation to make both [`park`] and
+/// [`unpark`] NOPs that return immediately.  Being unblocked does not imply
```

Comment:
`s/NOPs that return immediately/return immediately/`.

Reply:
In general since this is a public API, I would prefer wording to not be in terms of how the implementation could be (i.e. implemented to return immediately), but what the implementation is allowed to do (return spuriously without necessarily synchronizing with anything).

Reply:
I think examples are very useful to demonstrate where spurious wakeups may come from.  For me personally, I have seen this "spurious wakeups allowed" stuff often, but only when I recently read "NOPs are a valid implementation" *then* it clicked for me.

Reply:
I have changed the order a bit, putting the API contract first and the example second.  What do you think?